src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java

```diff
@@ -1904,4 +1907,55 @@ public AxisAlignedBB getCollisionBoundingBoxFromPool(World aWorld, int aX, int a
     public void onEntityCollidedWithBlock(World aWorld, int aX, int aY, int aZ, Entity collider) {
         mMetaTileEntity.onEntityCollidedWithBlock(aWorld, aX, aY, aZ, collider);
     }
+    
+    /**
+     * Shifts the machine Inventory index according to the change in Input/Output Slots.
+     * This is NOT done automatically. If you want to change slot count for a machine this method needs to be adapted.
+     * Currently this method only works for GT_MetaTileEntity_BasicMachine
+     * @param slotIndex The original Inventory index
+     * @param nbtVersion The GregTech version in which the original Inventory Index was saved.
+     * @return The corrected Inventory index
+     */
+    private int shiftInventoryIndex(int slotIndex, int nbtVersion){
+    	int oldInputSize, newInputSize, oldOutputSize, newOutputSize;
+    	int chemistryUpdateVersion = GT_Mod.calculateTotalGTVersion(509, 31);
+    	if (mID >= 211 && mID <= 218) {//Assembler
+    		if (nbtVersion < chemistryUpdateVersion) {
+    			oldInputSize = 2;
+    			oldOutputSize = 1;
+    		} else {
+    			return slotIndex;
+    		}
+    		newInputSize = 6;
+    		newOutputSize = 1;
+    	} else if (mID >= 421 && mID <= 428){//Chemical Reactor
+    		if (nbtVersion < chemistryUpdateVersion) {
+    			oldInputSize = 2;
+    			oldOutputSize = 1;
+    		} else {
+    			return slotIndex;
+    		}
+    		newInputSize = 2;
+    		newOutputSize = 2;
+    	} else if (mID >= 531 && mID <= 538) {//Distillery
+    		if (nbtVersion < chemistryUpdateVersion) {
+    			oldInputSize = 1;
+    			oldOutputSize = 0;
+    		} else {
+    			return slotIndex;
+    		}
+    		newInputSize = 1;
+    		newOutputSize = 1;
+    	} else {
+        	return slotIndex;    		
+    	}
+    	int indexShift = 0;
+    	if (slotIndex >= GT_MetaTileEntity_BasicMachine.OTHER_SLOT_COUNT + oldInputSize) {
```

Comment:
I decided to reference the constant so that it's immediately apparent why 4 is being added.
I think hard-coding the value would make the code more difficult to understand.